Mathlib/Analysis/Convex/Continuous.lean

```diff
@@ -0,0 +1,208 @@
+
+lemma ConvexOn.locallyLipschitzOn_iff_continuousOn (hC : IsOpen C) (hf : ConvexOn ℝ C f) :
+    LocallyLipschitzOn C f ↔ ContinuousOn f C := by
+  obtain rfl | hC' := C.eq_empty_or_nonempty
+  · simp
+  · exact (hf.continuousOn_tfae hC hC').out 0 1
+
+lemma ConcaveOn.locallyLipschitzOn_iff_continuousOn (hC : IsOpen C) (hf : ConcaveOn ℝ C f) :
+    LocallyLipschitzOn C f ↔ ContinuousOn f C := by
+  simpa using hf.neg.locallyLipschitzOn_iff_continuousOn hC
+
+protected lemma ConvexOn.locallyLipschitzOn (hC : IsOpen C) (hf : ConvexOn ℝ C f) :
+    LocallyLipschitzOn C f := by
+  obtain rfl | ⟨x₀, hx₀⟩ := C.eq_empty_or_nonempty
+  · simp
+  · obtain ⟨b, hx₀b, hbC⟩ := exists_mem_interior_convexHull_affineBasis (hC.mem_nhds hx₀)
+    refine ((hf.continuousOn_tfae hC ⟨x₀, hx₀⟩).out 3 0).mp ?_
+    refine ⟨x₀, hx₀, BddAbove.isBoundedUnder (IsOpen.mem_nhds isOpen_interior hx₀b) ?_⟩
+    exact (hf.bddAbove_convexHull ((subset_convexHull ..).trans hbC)
+      ((finite_range _).image _).bddAbove).mono (by gcongr; exact interior_subset)
+
+protected lemma ConcaveOn.locallyLipschitzOn (hC : IsOpen C) (hf : ConcaveOn ℝ C f) :
+    LocallyLipschitzOn C f := by simpa using hf.neg.locallyLipschitzOn hC
+
+protected lemma ConvexOn.continuousOn (hC : IsOpen C) (hf : ConvexOn ℝ C f) :
+    ContinuousOn f C := (hf.locallyLipschitzOn hC).continuousOn
+
+protected lemma ConcaveOn.continuousOn (hC : IsOpen C) (hf : ConcaveOn ℝ C f) :
+    ContinuousOn f C := (hf.locallyLipschitzOn hC).continuousOn
+
+lemma ConvexOn.locallyLipschitzOn_interior (hf : ConvexOn ℝ C f) :
+    LocallyLipschitzOn (interior C) f :=
+  (hf.subset interior_subset hf.1.interior).locallyLipschitzOn isOpen_interior
+
+lemma ConcaveOn.locallyLipschitzOn_interior (hf : ConcaveOn ℝ C f) :
+    LocallyLipschitzOn (interior C) f :=
+  (hf.subset interior_subset hf.1.interior).locallyLipschitzOn isOpen_interior
+
+lemma ConvexOn.continuousOn_interior (hf : ConvexOn ℝ C f) : ContinuousOn f (interior C) :=
+  hf.locallyLipschitzOn_interior.continuousOn
+
+lemma ConcaveOn.continuousOn_interior (hf : ConcaveOn ℝ C f) : ContinuousOn f (interior C) :=
+  hf.locallyLipschitzOn_interior.continuousOn
+
+protected lemma ConvexOn.locallyLipschitz (hf : ConvexOn ℝ univ f) : LocallyLipschitz f := by
+  simpa using hf.locallyLipschitzOn_interior
+
+protected lemma ConcaveOn.locallyLipschitz (hf : ConcaveOn ℝ univ f) : LocallyLipschitz f := by
+  simpa using hf.locallyLipschitzOn_interior
+
+-- proof_wanted ConvexOn.locallyLipschitzOn_intrinsicInterior (hf : ConvexOn ℝ C f) :
+--     ContinuousOn f (intrinsicInterior ℝ C)
+
+-- proof_wanted ConcaveOn.locallyLipschitzOn_intrinsicInterior (hf : ConcaveOn ℝ C f) :
+--     ContinuousOn f (intrinsicInterior ℝ C)
+
+-- proof_wanted ConvexOn.continuousOn_intrinsicInterior (hf : ConvexOn ℝ C f) :
+--     ContinuousOn f (intrinsicInterior ℝ C)
+
+-- proof_wanted ConcaveOn.continuousOn_intrinsicInterior (hf : ConcaveOn ℝ C f) :
+--     ContinuousOn f (intrinsicInterior ℝ C)
```

Comment:
Just noting that those `proof_wanted` are linted against by a combination of the unused arguments linter and the new variable inclusion mechanism linter!

cc @Kha

Reply:
That command should probably just disable those linters locally

Reply:
Opened https://github.com/leanprover-community/batteries/issues/938